### PR TITLE
ST: Fix several connect tests where race with NP was hit

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Bugfix

### Description
This PR fixes tests, where network policy was setup wrongly. All tests with connect or connects2i was affected by this issue.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST test_only crc

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST test_only crc

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST test_only crc

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST,ConnectS2I,TracingST#testProducerConsumerMirrorMakerConnectStreamsService+testConnectS2IService+testConnectService, test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST,ConnectS2I,TracingST#testProducerConsumerMirrorMakerConnectStreamsService+testConnectS2IService+testConnectService test_only crc

**Comment by Frawless** [comment]
After offline discussion with @see-quick it seems that failures are connected to some environment issues. I run all tests on my local env and it's working as expected. I think we should merge it and eventually fix failures in another PR.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectResource.java [hunk 1]
```diff
@@ -92,15 +88,15 @@ private static KafkaConnectBuilder defaultKafkaConnect(KafkaConnect kafkaConnect
             .endSpec();
     }
 
-    private static DoneableKafkaConnect deployKafkaConnect(KafkaConnect kafkaConnect, boolean allowNetworkPolicyAccess, String clusterName) {
+    private static DoneableKafkaConnect deployKafkaConnect(KafkaConnect kafkaConnect, String clusterName) {
+        if (Environment.DEFAULT_TO_DENY_NETWORK_POLICIES.equals("true")) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectResource.java]: "true" -> Boolean.TRUE.toString()

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectS2IResource.java [hunk 2]
```diff
@@ -75,15 +68,15 @@ public static KafkaConnectS2IBuilder defaultKafkaConnectS2I(KafkaConnectS2I kafk
             .endSpec();
     }
 
-    private static DoneableKafkaConnectS2I deployKafkaConnectS2I(KafkaConnectS2I kafkaConnectS2I, boolean allowNetworkPolicyAccess, String clusterName) {
+    private static DoneableKafkaConnectS2I deployKafkaConnectS2I(KafkaConnectS2I kafkaConnectS2I, String clusterName) {
+        if (Environment.DEFAULT_TO_DENY_NETWORK_POLICIES.equals("true")) {
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaConnectS2IResource.java]: same would be great to use static factory method `Boolean.TRUE.toString()`

**Pull request merged**